File: frontend/types/analytics.ts
export interface TimeSeriesPoint {
  date: string;
  value: number;
}

export interface CategorySales {
  category: string;
  revenue: number;
}

export interface ProductSales {
  product_id: string;
  name: string;
  quantity_sold: number;
  revenue: number;
}

export interface BundleSales {
  bundle_id: string;
  name: string;
  quantity_sold: number;
  revenue: number;
}

export interface InventoryStatusItem {
  product_id: string;
  name: string;
  variant: string | null;
  stock: number;
}

export interface Dashboard {
  registered_parents: number;
  total_players: number;
  total_orders: number;
  total_revenue: number;
  paid_revenue: number;
  pending_payments: number;
  pending_fulfillment: number;
  total_items_sold: number;
  revenue_over_time: TimeSeriesPoint[];
  orders_over_time: TimeSeriesPoint[];
  sales_by_category: CategorySales[];
  top_products: ProductSales[];
  bundle_sales: BundleSales[];
  inventory_status: InventoryStatusItem[];
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class TimeSeriesPoint(BaseModel):
    date: str
    value: float


class CategorySales(BaseModel):
    category: str
    revenue: float


class ProductSales(BaseModel):
    product_id: str
    name: str
    quantity_sold: int
    revenue: float


class BundleSales(BaseModel):
    bundle_id: str
    name: str
    quantity_sold: int
    revenue: float


class InventoryStatusItem(BaseModel):
    product_id: str
    name: str
    variant: str | None = None
    stock: int


class DashboardResponse(BaseModel):
    registered_parents: int
    total_players: int
    total_orders: int
    total_revenue: float
    paid_revenue: float
    pending_payments: int
    pending_fulfillment: int
    total_items_sold: int
    revenue_over_time: list[TimeSeriesPoint]
    orders_over_time: list[TimeSeriesPoint]
    sales_by_category: list[CategorySales]
    top_products: list[ProductSales]
    bundle_sales: list[BundleSales]
    inventory_status: list[InventoryStatusItem]
